src/core/layout/qgslayoutitemelevationprofile.cpp

```diff
@@ -0,0 +1,421 @@
+/***************************************************************************
+                             qgslayoutitemelevationprofile.cpp
+                             ---------------------------------
+    begin                : January 2023
+    copyright            : (C) 2023 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgslayoutitemelevationprofile.h"
+#include "qgslayoutitemregistry.h"
+#include "qgsplot.h"
+#include "qgsmessagelog.h"
+
+///@cond PRIVATE
+class QgsLayoutItemElevationProfilePlot : public Qgs2DPlot
```

Comment:
@nyalldawson , I'll be very interested in how you develop this specific class :) I think this is one place where we could share code between the qgsquick stuff and layout.